lib/ansible/cli/config.py

```diff
@@ -553,17 +571,60 @@ def _get_plugin_configs(self, ptype, plugins):
 
         return output
 
+    def _get_galaxy_server_configs(self):
+
+        output = []
+        # add galaxy servers
+        for server in self._galaxy_servers:
+            server_config = {}
+            s_config = self.config.get_configuration_definitions('galaxy_server', server)
+            for setting in s_config.keys():
+                try:
+                    v, o = C.config.get_config_value_and_origin(setting, plugin_type='galaxy_server', plugin_name=server, cfile=self.config_file)
+                except AnsibleError as e:
+                    if to_text(e).startswith('No setting was provided for required configuration'):
```

Comment:
This seems brittle, any chance we can pass a sentinel value as `direct` (can't remember where that sits in the precedence hierarchy) or expose a way to provide a default/fallback sentinel value to `get_config_value_and_origin` to check if the required value was set or not?

Reply:
direct has highest precedence and since we get origin, we know how it was set, no need to check against default/sentinel or not

Reply:
The sentinel value was more another way to determine that no value was set for a required option. It sounds like using direct is out but I don’t think catching the generic AnsibleError and parsing the exception output is great. Could we add a `default` option that we could use to identify this instead?

Reply:
no since if we add a default, it is not required, the config system considers those 2 in opposition

Reply:
I think moving to a specific error class would resolve this